src/formatter-settings/index.ts

```diff
@@ -1,99 +1,306 @@
+            const err: Error = new Error(`The extension version of 'redhat.java' (${javaExtVersion}) is too stale.`);
+            setUserError(err);
+            sendError(err);
+            return false;
+        }
+        await javaExt.activate();
+        this.checkedRequirement = true;
+        return true;
     }
 
-    private async formatWithProfileSettings(webviewPanel: vscode.WebviewPanel) {
-        // use default settings temporarily
-        const content = await vscode.commands.executeCommand<string>("java.execute.workspaceCommand", "java.edit.stringFormatting", Example.getExample(this.exampleKind), JSON.stringify([]), JavaConstants.CURRENT_FORMATTER_SETTINGS_VERSION);
-        if (webviewPanel && webviewPanel.webview) {
-            webviewPanel.webview.postMessage({
+    private async initialize(document: vscode.TextDocument): Promise<boolean> {
+        if (!await this.checkRequirement()) {
+            return false;
+        }
+        if (!await this.checkProfile()) {
+            return false;
+        }
+        this.exampleKind = ExampleKind.INDENTATION_EXAMPLE;
+        const onDidChangeSetting: vscode.Disposable = vscode.workspace.onDidChangeConfiguration(async (e) => {
+            if (e.affectsConfiguration(`java.${JavaConstants.SETTINGS_URL_KEY}`) || e.affectsConfiguration(`java.${JavaConstants.SETTINGS_PROFILE_KEY}`)) {
+                this.onChangeProfile();
+            } else if (e.affectsConfiguration(VSCodeSettings.TAB_SIZE) || e.affectsConfiguration(VSCodeSettings.INSERT_SPACES) || e.affectsConfiguration(VSCodeSettings.DETECT_INDENTATION)) {
+                await this.updateVSCodeSettings();
+                this.format();
+            }
+        });
+        this.webviewPanel?.onDidDispose(() => {
+            onDidChangeSetting.dispose();
+        });
+        await this.parseProfileAndUpdate(document);
+        return true;
+    }
+
+    private async parseProfileAndUpdate(document: vscode.TextDocument): Promise<void> {
+        const content: ProfileContent = parseProfile(document);
+        this.diagnosticCollection.set(document.uri, content.diagnostics);
+        if (this.webviewPanel) {
+            this.profileElements = content.profileElements || this.profileElements;
+            this.profileSettings = content.profileSettings || this.profileSettings;
+            this.lastElement = content.lastElement || this.lastElement;
+            this.settingsVersion = content.settingsVersion;
+            if (content.supportedProfileSettings) {
+                this.webviewPanel.webview.postMessage({
+                    command: "loadProfileSetting",
+                    setting: Array.from(content.supportedProfileSettings.values()),
+                });
+            }
+            await this.updateVSCodeSettings();
+            this.format();
+        }
+    }
+
+    private onChangeProfile(): void {
```

Comment:
We need to distinguish profile and the setting pointing to the profile location.

`onChangeProfile()` sounds like the event when the profile file is changed

Reply:
will change the ambiguity names.